### PR TITLE
[fix](function) the bucket number arg of width_bucket should be a positive integer value

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Review by suxiaogang223 — APPROVED** [review]

File: be/src/vec/functions/function_width_bucket.cpp [hunk 1]
```diff
@@ -86,12 +92,12 @@ class FunctionWidthBucket : public IFunction {
     void _execute(const IColumn& expr_column, const IColumn& min_value_column,
                   const IColumn& max_value_column, const int64_t num_buckets,
                   IColumn& nested_column) const {
-        const ColumnType& expr_column_concrete = reinterpret_cast<const ColumnType&>(expr_column);
-        const ColumnType& min_value_column_concrete =
+        const auto& expr_column_concrete = reinterpret_cast<const ColumnType&>(expr_column);
```
Comment (zhiqiang-hhhh) [inline be/src/vec/functions/function_width_bucket.cpp:95]: why not using assert_cast?

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Pull request merged**